tests/pytests/test_dipoles.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+
+import qcelemental as qcel
+import psi4
+
+from .utils import compare_values
+
+perturbation_strength = 0.001
+
+@pytest.mark.slow
+# TODO: That "true" needs to be a string is silly. Convert it to a boolean when you can do that without incurring a NaN energy.
+@pytest.mark.parametrize("inp", [
+    pytest.param({'name': 'omp2', 'options': {'mp2_type': 'df', 'max_mograd_convergence': 6}, 'varname': 'DF-OMP2'}, id='df-omp2 ae'),
+    pytest.param({'name': 'omp2', 'options': {'mp2_type': 'df', 'freeze_core': 'true', 'max_mograd_convergence': 6}, 'varname': 'DF-OMP2'}, id='df-omp2 fc'),
+    pytest.param({'name': 'omp3', 'options': {'mp_type': 'df', 'max_mograd_convergence': 6}, 'varname': 'DF-OMP3'}, id='df-omp3 ae'),
+    pytest.param({'name': 'omp3', 'options': {'mp_type': 'df', 'freeze_core': 'true', 'max_mograd_convergence': 6}, 'varname': 'DF-OMP3'}, id='df-omp3 fc'),
+    pytest.param({'name': 'omp2.5', 'options': {'mp_type': 'df', 'max_mograd_convergence': 6}, 'varname': 'DF-OMP2.5'}, id='df-omp2.5 ae'),
+    pytest.param({'name': 'omp2.5', 'options': {'mp_type': 'df', 'freeze_core': 'true', 'max_mograd_convergence': 6}, 'varname': 'DF-OMP2.5'}, id='df-omp2.5 fc'),
+    pytest.param({'name': 'olccd', 'options': {'cc_type': 'df', 'max_mograd_convergence': 6}, 'varname': 'DF-OLCCD'}, id='df-olccd ae'),
+    pytest.param({'name': 'olccd', 'options': {'cc_type': 'df', 'freeze_core': 'true', 'max_mograd_convergence': 6}, 'varname': 'DF-OLCCD'}, id='df-olccd fc')
+    ]
+)
+def test_dipole(inp):
+    h2o = psi4.geometry("""
+        O
+        H 1 1.0
+        H 1 1.0 2 101.5
+    """)
+
+    psi4.set_options({'perturb_h': True, 'perturb_with': 'dipole', 'basis': 'cc-pvdz'})
+    psi4.set_options(inp['options'])
+    energies = []
+    for l in [1, -1, 2, -2]:
+        psi4.set_options({'perturb_dipole': [0, 0, l * perturbation_strength]})
+        energies.append(psi4.energy(inp['name']))
+    findif_dipole = (8 * energies[0] - 8 * energies[1] - energies[2] + energies[3]) / (12 * perturbation_strength) * qcel.constants.dipmom_au2debye
```

Comment:
So is this just `(8 * ((l = +1) - (l = -1)) - 1 * ((l = +2) - (l = -2)))`? The indices to `energies[]` got me confused.

Reply:
Correct. I'll be changing `energies` to a dict and using `(8 * energies[1] - 8 * energies[-1] - energies[2] + energies[-2])` in the next commit.